### PR TITLE
Fix Cypher generation iterative loop

### PR DESCRIPTION
**Pull request by galshubeli**
## Summary by CodeRabbit

- **New Features**
	- Added new terms "openai" and "Ollama" to the word list.
	- Introduced constants for enhanced ontology creation and data extraction processes.
	- Added methods to delete the last message in various chat session classes.

- **Bug Fixes**
	- Improved error handling and retry logic in the Graph Query Generation step.

- **Documentation**
	- Updated constants and methods to enhance clarity in ontology management.

- **Refactor**
	- Restructured attribute initialization in the Orchestrator class for better encapsulation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,7 +47,9 @@ docker run -p 6379:6379 -p 3000:3000 -it --rm  -v ./data:/data falkordb/falkordb
 Currently, this SDK supports the following LLMs API:
 
 * [OpenAI](https://openai.com/index/openai-api) Recommended model:`gpt-4o`
-* [google](https://makersuite.google.com/app/apikey) Recommended model:`gemini-1.5-flash-001`
+* [Google](https://makersuite.google.com/app/apikey) Recommended model:`gemini-1.5-flash-001`
```
Comment (swilly22) [inline README.md:50]: See my comments on the previous PR dealing with README / documentation changes

File: graphrag_sdk/fixtures/prompts.py [hunk 2]
```diff
@@ -441,9 +441,10 @@
 """
 
 CYPHER_GEN_PROMPT = """
-Using the ontology provided, generate an OpenCypher statement to query the graph database returning all relevant entities, relationships, and attributes to answer the question below:
+Using the ontology provided, generate an OpenCypher statement to query the graph database returning all relevant entities, relationships, and attributes to answer the question below.
 If you cannot generate a OpenCypher statement for any reason, return an empty string.
 Respect the order of the relationships, the arrows should always point from the "source" to the "target".
+Please think if your answer is a valid Cypher query, and correct it if it is not.
```
Comment (swilly22) [inline graphrag_sdk/fixtures/prompts.py:447]: This is what made the improvement ?

File: graphrag_sdk/models/azure_openai.py [hunk 3]
```diff
@@ -227,3 +227,22 @@ def _get_generation_config(self, output_method: OutputMethod):
             config['response_format'] = { "type": "json_object" }
         
         return config
+    
+    def delete_last_message(self):
+        """
+        Deletes the last message exchange (user message and assistant response) from the chat history.
+        Preserves the system message if present.
+
```
Comment (swilly22) [inline graphrag_sdk/models/azure_openai.py:235]: Provide an example of the chat history before and after

File: graphrag_sdk/models/azure_openai.py [hunk 4]
```diff
@@ -227,3 +227,22 @@ def _get_generation_config(self, output_method: OutputMethod):
...
+        """
+        Deletes the last message exchange (user message and assistant response) from the chat history.
+        Preserves the system message if present.
+        
+        Note: Does nothing if the chat history is empty or contains only a system message.
+        """
+        # Keep at least the system message if present
+        min_length = 1 if self._model.system_instruction else 0
+        if len(self._history) - 2 >= min_length:
+            self._history = self._history[:-2]
```
Comment (swilly22) [inline graphrag_sdk/models/azure_openai.py:241]: might be more efficient cheaper to just call: `self._history.pop()` twice removing the last two elements.

File: graphrag_sdk/models/ollama.py [hunk 5]
```diff
@@ -216,3 +216,22 @@ def _adjust_generation_config(self, output_method: OutputMethod) -> dict:
             config['format'] = 'json'
         
         return config
+    
+    def delete_last_message(self):
```
Comment (swilly22) [inline graphrag_sdk/models/ollama.py:220]: this is tedious, can't we have a chat history class serving all models ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: graphrag_sdk/fixtures/prompts.py (hunk 2 above)
Comment (galshubeli) [reply graphrag_sdk/fixtures/prompts.py:447]: Partially, it is less likely to happen. It is combined with an independent generation loop

**Pull request merged**